modules/users/tests/server/user-profile.server.routes.tests.js

```diff
@@ -568,6 +568,118 @@ describe('User profile CRUD tests', function () {
+      function (done) {
+        var user2 = _user;
+        delete user2.email;
+        agent.post('/api/auth/signin')
+          .send(credentials)
+          .expect(200)
+          .end(function (err) {
+            if (err) {
+              return done(err);
+            }
+            // First change
+            user2.username = _user.username + '01';
+            agent.put('/api/users')
+              .send(user2)
+              .expect(200)
+              .end(function (err, res) {
+                if (err) {
+                  return done(err);
+                }
+                res.body.username.should.equal(user2.username);
+                user2.username = _user.username + '02';
+                agent.put('/api/users')
+                  .send(user2)
+                  .end(function (err, res) {
+                    if (err) {
+                      return done(err);
+                    }
+                    res.body.message.should.equal('You have used your chance to change your username already');
+                    return done();
+                  });
+              });
+          });
+      });
+
+    it('should update if the three months rule have passed by',
+      function (done) {
+        var user2 = _user;
+        delete user2.email;
+        agent.post('/api/auth/signin')
+          .send(credentials)
+          .expect(200)
+          .end(function (err) {
+            if (err) {
+              return done(err);
+            }
+            // First change
+            user2.username = _user.username + '01';
+            agent.put('/api/users')
+              .send(user2)
+              .expect(200)
+              .end(function (err, res) {
+                if (err) {
+                  return done(err);
+                }
+                res.body.username.should.equal(user2.username);
+                User.findById(user._id, function (err, user) {
+                  if (err) {
+                    return done(err);
+                  }
+                  var threeMonthsAgo = new Date(user.usernameUpdated);
+                  threeMonthsAgo.setMonth(
+                    user.usernameUpdated.getMonth() - 3
+                  );
+                  user.update(
+                    { $set: { usernameUpdated: threeMonthsAgo } },
+                    function (err) {
+                      if (err) {
+                        return done(err);
+                      }
+                      user2.username = _user.username + '02';
+                      agent.put('/api/users')
+                        .send(user2)
+                        .end(function (err, res) {
+                          if (err) {
+                            return done(err);
+                          }
+                          res.body.username.should.equal(user2.username);
+                          return done();
+                        });
+                    });
+                });
+              });
+          });
+      });
+  });
+
```

Comment:
We need a test to confirm user cannot update `usernameUpdated` field on their own.

Reply:
line 707

Reply:
We can loop back and add the test in a follow-up PR. 

The concern indeed was that `usernameUpdated` would be secured well enough since there was an issue before with it. It should be fixed now as far as I can see but it's pretty critical value for security so tests :thumbsup: 👍 :thumbsup: